### PR TITLE
[Communicator] adding SynchronizeNodalFlags

### PR DESCRIPTION
**Pull request by philbucher**
as discussed

### CODE REVIEW

**Review by loumalouomega — DISMISSED** [review]
Thanks

**Comment by philbucher** [comment]
@loumalouomega please don't merge yet, @jcotela wanted to add a test

**Review by jcotela — APPROVED** [review]
@philbucher the test passes, so I'm approving. Please give it a look before merging, in case I did something stupid.

File: kratos/mpi/tests/sources/test_mpi_communicator.cpp [hunk 1]
```diff
@@ -178,6 +178,38 @@ KRATOS_TEST_CASE_IN_SUITE(MPICommunicatorSynchronizeAnd, KratosMPICoreFastSuite)
...
+    ModelPart& r_model_part = model.CreateModelPart("TestModelPart");
+    r_model_part.AddNodalSolutionStepVariable(PARTITION_INDEX);
+
+    MPIDataCommunicator world_comm(MPI_COMM_WORLD);
+    Internals::ModelPartForMPICommunicatorTests(r_model_part, world_comm);
+
+    const int rank = world_comm.Rank();
+    const bool rank_is_even( (rank % 2) == 0 );
+
+    for (auto i_node = r_model_part.NodesBegin(); i_node != r_model_part.NodesEnd(); ++i_node)
```
Comment (philbucher) [inline kratos/mpi/tests/sources/test_mpi_communicator.cpp:193]: @jcotela why not adding it only on local nodes?

Comment (jcotela) [reply kratos/mpi/tests/sources/test_mpi_communicator.cpp:193]: It probably makes no difference, but in the test ModelPart one of the ghost nodes is always in rank i+1 (the other is the center node, on rank 0). Setting it to the local value (true if i is even) makes sure that the ghost node is set at the wrong value, and that the test must fail if communication goes wrong.

Comment (philbucher) [reply kratos/mpi/tests/sources/test_mpi_communicator.cpp:193]: ok :+1:

**Pull request merged**